SUS-2178 | no need to pass $this by a reference

macbre opened the pull request:
https://wikia-inc.atlassian.net/browse/SUS-2178

Objects are passed by the reference since PHP5. There's no need to make it explicit via `&$this` syntax and it makes PHP 7.1 complain.

https://bugs.php.net/bug.php?id=73751

https://3v4l.org/vMc1N contains the sample code where `&$this` as part of the callback invocation:

```php
Hooks::run( 'foo', array( &$this, &$var ) );
```

causes a "Warning: Parameter 1 to {closure}() expected to be a reference, value given" on 7.1.0.

Running the provided sample code on 5.3.0 - 5.6.25, hhvm-3.10.0 - 3.12.0, 7.0.0 - 7.0.10 works as expected.

Re-declaring $this will pass on 7.1.0 and 5.3.0 - 5.6.25, hhvm-3.10.0 - 3.12.0, 7.0.0 - 7.0.10.

macbre committed:
SUS-2178 | no need to pass $this with a reference

Objects are passed by the reference since PHP5.

https://bugs.php.net/bug.php?id=73751

https://3v4l.org/vMc1N contains the sample code where `&$this` as part of the callback invocation:

```php
Hooks::run( 'foo', array( &$this, &$var ) );
```

causes a "Warning: Parameter 1 to {closure}() expected to be a reference, value given" on 7.1.0.

Running the provided sample code on 5.3.0 - 5.6.25, hhvm-3.10.0 - 3.12.0, 7.0.0 - 7.0.10 works as expected.

Re-declaring $this will pass on 7.1.0 and 5.3.0 - 5.6.25, hhvm-3.10.0 - 3.12.0, 7.0.0 - 7.0.10.

WikiaTech commented on lib/vendor/PEAR/Command/Channels.php:
```diff
@@ -164,7 +164,7 @@ function doList($command, $options, $params)
         $registered = $reg->getChannels();
-        usort($registered, array(&$this, '_sortchannels'));
+        usort($registered, array($this, '_sortchannels'));
         $i = $j = 0;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$i" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$j" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on includes/OutputPage.php:
```diff
@@ -2127,14 +2127,14 @@ public function output() {
 		}
 
-		wfRunHooks( 'BeforeSendCacheControl', array( &$this ) ); // Wikia change
+		wfRunHooks( 'BeforeSendCacheControl', array( $this ) ); // Wikia change
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/SemanticForms/specials/SF_UploadWindow.php:
```diff
@@ -327,7 +327,7 @@ protected function processUpload() {
 		if ( !$status->isOK() )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/SemanticForms/specials/SF_UploadWindow.php:
```diff
@@ -428,7 +428,7 @@ protected function processUpload() {
 		print $output;
 		$img = null; // @todo: added to avoid passing a ref to null - should this be defined somewhere?
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$img" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php:
```diff
@@ -91,7 +91,7 @@ public function Notification() {
 		$msg = "";
 		wfRunHooks( 'NotificationGetNotificationMessage', [
-			&$this, &$msg, $firstEntity->isMain(), $data, $authors, $userCount, $wg->User->getName()
+			$this, &$msg, $firstEntity->isMain(), $data, $authors, $userCount, $wg->User->getName()
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on includes/EditPage.php:
```diff
@@ -2081,7 +2081,7 @@ protected function showHeader() {
 						/* Wikia change begin - @author: Marooned */
 						/* Add new hook and condition to allow alternate message to be displayed when editing old revision */
-						if ( wfRunHooks( 'EditPage::showEditForm:oldRevisionNotice', array( &$this ) ) ) {
+						if ( wfRunHooks( 'EditPage::showEditForm:oldRevisionNotice', array( $this ) ) ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

WikiaTech commented on includes/EditPage.php:
```diff
@@ -1878,14 +1878,14 @@ function showEditForm( $formCallback = null ) {
 
 		// wikia change begin
-		wfRunHooks ('EditForm:BeforeDisplayingTextbox', array (&$this, &$hidden /* TODO: remove? */) ) ;
+		wfRunHooks ('EditForm:BeforeDisplayingTextbox', array ($this, &$hidden /* TODO: remove? */) ) ;
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech replied:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on includes/EditPage.php:
```diff
@@ -1910,7 +1910,7 @@ function showEditForm( $formCallback = null ) {
 
 		// wikia change begin
-		wfRunHooks ('EditForm:AfterDisplayingTextbox', array (&$this, &$hidden /* TODO: remove? */) ) ;
+		wfRunHooks ('EditForm:AfterDisplayingTextbox', array ($this, &$hidden /* TODO: remove? */) ) ;
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech replied:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on includes/EditPage.php:
```diff
@@ -3161,7 +3161,7 @@ function noSuchSectionPage() {
 		$wgOut->prepareErrorPage( wfMessage( 'nosuchsectiontitle' ) );
 
 		$res = wfMsgExt( 'nosuchsectiontext', 'parse', $this->section );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

WikiaTech commented on includes/EditPage.php:
```diff
@@ -2556,7 +2556,7 @@ protected function showStandardInputs( &$tabindex = 2 ) {
 	 */
 	protected function showConflict() {
 		global $wgOut;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on includes/WikiPage.php:
```diff
@@ -1967,7 +1967,7 @@ public function doUpdateRestrictions( array $limit, array $expiry, &$cascade, $r
 			wfRunHooks( 'NewRevisionFromEditComplete', array( $this, $nullRevision, $latest, $user ) );
-			wfRunHooks( 'ArticleProtectComplete', array( &$this, &$user, $limit, $reason ) );
+			wfRunHooks( 'ArticleProtectComplete', array( $this, &$user, $limit, $reason ) );
 		} else { # Protection of non-existing page (also known as "title protection")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on includes/WikiPage.php:
```diff
@@ -1745,7 +1745,7 @@ public function doEditUpdates( Revision $revision, User $user, array $options =
 			&& !( $revision->isMinor() && $user->isAllowed( 'nominornewtalk' ) )
 		) {
-			if ( wfRunHooks( 'ArticleEditUpdateNewTalk', array( &$this ) ) ) {
+			if ( wfRunHooks( 'ArticleEditUpdateNewTalk', array( $this ) ) ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

WikiaTech commented on includes/WikiPage.php:
```diff
@@ -1907,7 +1907,7 @@ public function doUpdateRestrictions( array $limit, array $expiry, &$cascade, $r
 		$protectDescription = trim( $protectDescription );
 
 		if ( $id ) { # Protection of existing page
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on includes/WikiPage.php:
```diff
@@ -1701,9 +1701,9 @@ public function doEditUpdates( Revision $revision, User $user, array $options =
 
-		if ( wfRunHooks( 'ArticleEditUpdatesDeleteFromRecentchanges', array( &$this ) ) ) {
+		if ( wfRunHooks( 'ArticleEditUpdatesDeleteFromRecentchanges', array( $this ) ) ) {
 			if ( 0 == mt_rand( 0, 99 ) ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

WikiaTech commented on includes/WikiPage.php:
```diff
@@ -2204,7 +2204,7 @@ public function doDeleteArticleReal(
 		$hookAddedLogEntry = false;
 
 		# @todo mediawiki merge 1.19: $log doesn't exist there
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on includes/db/DatabaseOracle.php:
```diff
@@ -884,7 +884,7 @@ function tableExists( $table, $fname = __METHOD__ ) {
 	private function fieldInfoMulti( $table, $field ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/LatestPhotos/LatestPhotosHooks.class.php:
```diff
@@ -13,7 +13,7 @@ public static function onImageUploadComplete( &$image ){
 	}
 
-	public static function onImageRenameCompleated( &$this , &$ot , &$nt ){
+	public static function onImageRenameCompleated( $this , &$ot , &$nt ){
```
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this use of "$this". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2014)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$this". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$ot". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/SemanticForms/includes/SF_Template.php:
```diff
@@ -344,7 +344,7 @@ public function createCargoStoreCall() {
 	 * extension.
 	 */
 	public function createText() {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "createText" is 67 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "createText" has 299 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "createText" is 67 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "createText" has 299 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

WikiaTech commented on extensions/wikia/Blogs/SpecialCreateBlogListingPage.php:
```diff
@@ -25,7 +25,7 @@ public function __construct() {
 	public function execute( $par ) {
 		global $wgOut, $wgUser, $wgRequest, $wgTitle;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$wgTitle" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on lib/vendor/PEAR/Builder.php:
```diff
@@ -106,7 +106,7 @@ function _build_win32($descfile, $callback = null)
         $command = 'msdev '.$dsp.' /MAKE "'.$info['package']. ' - Release"';
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$info" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on includes/parser/Tidy.php:
```diff
@@ -42,7 +42,7 @@ public function getWrapped( $text ) {
-			array( &$this, 'replaceEditSectionLinksCallback' ), $text );
+			array( $this, 'replaceEditSectionLinksCallback' ), $text );
 
 		$wrappedtext = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"'.
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$wrappedtext". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on lib/vendor/PEAR/Config.php:
```diff
@@ -607,7 +607,7 @@ function __construct($user_file = '', $system_file = '', $ftp_file = false,
         $this->_registry['default']->setConfig($this);
         $this->_regInitialized['default'] = false;
-        //$GLOBALS['_PEAR_Config_instance'] = &$this;
+        //$GLOBALS['_PEAR_Config_instance'] = $this;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech commented on lib/vendor/PEAR/Config.php:
```diff
@@ -607,7 +607,7 @@ function __construct($user_file = '', $system_file = '', $ftp_file = false,
+        //$GLOBALS['_PEAR_Config_instance'] = $this;
     }
 
     // }}}
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech commented on includes/Article.php:
```diff
@@ -509,7 +509,7 @@ public function view() {
-					wfRunHooks( 'ArticleViewHeader', array( &$this, &$outputDone, &$useParserCache ) );
+					wfRunHooks( 'ArticleViewHeader', array( $this, &$outputDone, &$useParserCache ) );
 					break;
 				case 2:
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this "switch/case" number of lines from 56 to at most 10, for example by extracting code into function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this "switch/case" number of lines from 56 to at most 10, for example by extracting code into function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)

WikiaTech commented on includes/Article.php:
```diff
@@ -1371,7 +1371,7 @@ public function delete() {
 		global $wgOut, $wgRequest, $wgLang;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on includes/Article.php:
```diff
@@ -509,7 +509,7 @@ public function view() {
 		while ( !$outputDone && ++$pass ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS881)

WikiaTech commented on includes/Article.php:
```diff
@@ -1165,7 +1165,7 @@ public function showDeletedRevisionHeader() {
 	public function setOldSubtitle( $oldid = 0 ) {
 		global $wgLang, $wgOut, $wgUser, $wgRequest;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on includes/Article.php:
```diff
@@ -350,7 +350,7 @@ function fetchContent() {
 		$this->mContent = $this->mRevision->getText( Revision::FOR_THIS_USER ); // Loads if user is allowed
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on includes/specials/SpecialVersion.php:
```diff
@@ -294,7 +294,7 @@ function getExtensionCredits() {
-		wfRunHooks( 'SpecialVersionExtensionTypes', array( &$this, &$extensionTypes ) );
+		wfRunHooks( 'SpecialVersionExtensionTypes', array( $this, &$extensionTypes ) );
 
 		$out = Xml::element( 'h2', array( 'id' => 'mw-version-ext' ), wfMessage( 'version-extensions' )->plain() ) .
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change the output format of this i18n string away from plain(). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3APlainFormatWfMessage)

WikiaTech commented on includes/Title.php:
```diff
@@ -1508,15 +1508,15 @@ public function getLocalURL( $query = '', $query2 = false ) {
-			wfRunHooks( 'GetLocalURL::Internal', array( &$this, &$url, $query ) );
+			wfRunHooks( 'GetLocalURL::Internal', array( $this, &$url, $query ) );
 
 			// @todo FIXME: This causes breakage in various places when we
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on lib/vendor/PEAR/Installer.php:
```diff
@@ -1524,7 +1524,7 @@ function sortPackagesForUninstall(&$packages)
+        usort($packages, array($this, '_sortUninstall'));
     }
 
     function _sortUninstall($a, $b)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on maintenance/dumpTextPass.php:
```diff
@@ -269,8 +269,8 @@ function readDump( $input ) {
+		xml_set_element_handler( $parser, array( $this, 'startElement' ), array( $this, 'endElement' ) );
+		xml_set_character_data_handler( $parser, array( $this, 'characterData' ) );
 
 		$offset = 0; // for context extraction on error reporting
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on includes/parser/Parser.php:
```diff
@@ -5423,7 +5423,7 @@ function makeImage( $title, $options, $holders = false ) {
 		# wikia start
-		if (!wfRunHooks( 'BeforeParserMakeImageLinkObjOptions', array( &$this, &$title, &$parts, &$params, &$time, &$descQuery, $options ) ) ) {
+		if (!wfRunHooks( 'BeforeParserMakeImageLinkObjOptions', array( $this, &$title, &$parts, &$params, &$time, &$descQuery, $options ) ) ) {
 			$ret = Linker::makeImageLink2( $title, $file, $params['frame'], $params['handler'], $time, $descQuery );
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$ret". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on includes/parser/Parser.php:
```diff
@@ -2867,14 +2867,14 @@ function getVariableValue( $index, $frame = false ) {
 		 */
-		if ( wfRunHooks( 'ParserGetVariableValueVarCache', array( &$this, &$this->mVarCache ) ) ) {
+		if ( wfRunHooks( 'ParserGetVariableValueVarCache', array( $this, &$this->mVarCache ) ) ) {
 			if ( isset( $this->mVarCache[$index] ) ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

WikiaTech commented on includes/User.php:
```diff
@@ -1516,7 +1516,7 @@ public function isPingLimitable() {
 	public function pingLimiter( $action = 'edit' ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "pingLimiter" is 23 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on maintenance/cleanupTitles.php:
```diff
@@ -76,7 +76,7 @@ protected function fileExists( $name ) {
-			array( &$this, 'hexChar' ),
+			array( $this, 'hexChar' ),
 			$row->page_title );
 		if ( $legalized == '.' ) $legalized = '(dot)';
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on maintenance/cleanupTitles.php:
```diff
@@ -76,7 +76,7 @@ protected function fileExists( $name ) {
+			array( $this, 'hexChar' ),
 			$row->page_title );
 		if ( $legalized == '.' ) $legalized = '(dot)';
 		if ( $legalized == '_' ) $legalized = '(space)';
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on includes/specials/SpecialMovepage.php:
```diff
@@ -493,7 +493,7 @@ function doSubmit() {
+		wfRunHooks( 'SpecialMovepageAfterMove', array( $this, &$ot, &$nt ) );
 
 		$out = $this->getOutput();
 		$out->setPagetitle( wfMsg( 'pagemovedsub' ) );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

WikiaTech commented on lib/vendor/PEAR/Command/Registry.php:
```diff
@@ -140,7 +140,7 @@ function doList($command, $options, $params)
         $installed = $reg->packageInfo(null, null, $channel);
-        usort($installed, array(&$this, '_sortinfo'));
+        usort($installed, array($this, '_sortinfo'));
         $i = $j = 0;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$i" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$j" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

WikiaTech commented:
SonarQube analysis reported 3626 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 2391 major
* ![MINOR][MINOR] 331 minor
* ![INFO][INFO] 897 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [cloudfiles_http.php#L339](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php#L339): Move arguments "$ttl", "$cdn_log_retention", "$cdn_acl_user_agent" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Parser.php#L4986](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/includes/parser/Parser.php#L4986): Move arguments "$title" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Parser.php#L4996](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/includes/parser/Parser.php#L4996): Move arguments "$title" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Language.php#L3639](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/languages/Language.php#L3639): Move arguments "$prefix" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [PEAR.php#L909](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/lib/vendor/PEAR.php#L909): Remove this use of the "eval" function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)
1. ![CRITICAL][CRITICAL] [Installer.php#L344](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/lib/vendor/PEAR/Installer.php#L344): Remove this use of the "eval" function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)
1. ![MAJOR][MAJOR] [HasCategory.php#L15](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/extensions/3rdparty/HasCategory/HasCategory.php#L15): Remove the parentheses from this "echo" call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2041)
1. ![MAJOR][MAJOR] [HasCategory.php#L34](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/extensions/3rdparty/HasCategory/HasCategory.php#L34): Do not use the global wgParser Parser instance. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWgParserUsage)
1. ![MAJOR][MAJOR] [HasCategory.php#L97](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/extensions/3rdparty/HasCategory/HasCategory.php#L97): Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [HasCategory.php#L104](https://github.com/Wikia/app/blob/bf992b8e20e3209ee64d4e670719724e0524e95d/extensions/3rdparty/HasCategory/HasCategory.php#L104): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
SUS-2178 | remove references for object method arguments

Causes Warning: Parameter 1 to Captcha\Hooks::APIGetAllowedParams() expected to be a reference, value given in /includes/Hooks.php on line 216

WikiaTech commented on extensions/VisualEditor/VisualEditor.hooks.php:
```diff
@@ -450,7 +450,7 @@ public static function onResourceLoaderGetConfigVars( array &$vars ) {
 	 */
-	public static function onResourceLoaderRegisterModules( ResourceLoader &$resourceLoader ) {
+	public static function onResourceLoaderRegisterModules( ResourceLoader $resourceLoader ) {
 		global $wgResourceModules;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$wgResourceModules" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

WikiaTech commented:
SonarQube analysis reported 3670 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 2428 major
* ![MINOR][MINOR] 331 minor
* ![INFO][INFO] 904 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [cloudfiles_http.php#L339](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php#L339): Move arguments "$ttl", "$cdn_log_retention", "$cdn_acl_user_agent" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Parser.php#L4986](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/includes/parser/Parser.php#L4986): Move arguments "$title" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Parser.php#L4996](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/includes/parser/Parser.php#L4996): Move arguments "$title" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [Language.php#L3639](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/languages/Language.php#L3639): Move arguments "$prefix" after arguments without default value [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![CRITICAL][CRITICAL] [PEAR.php#L909](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/lib/vendor/PEAR.php#L909): Remove this use of the "eval" function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)
1. ![CRITICAL][CRITICAL] [Installer.php#L344](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/lib/vendor/PEAR/Installer.php#L344): Remove this use of the "eval" function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)
1. ![MAJOR][MAJOR] [HasCategory.php#L15](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/extensions/3rdparty/HasCategory/HasCategory.php#L15): Remove the parentheses from this "echo" call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2041)
1. ![MAJOR][MAJOR] [HasCategory.php#L34](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/extensions/3rdparty/HasCategory/HasCategory.php#L34): Do not use the global wgParser Parser instance. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWgParserUsage)
1. ![MAJOR][MAJOR] [HasCategory.php#L97](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/extensions/3rdparty/HasCategory/HasCategory.php#L97): Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [HasCategory.php#L104](https://github.com/Wikia/app/blob/f1075b63d5a718662f4d30c2af1c7be0ef99c067/extensions/3rdparty/HasCategory/HasCategory.php#L104): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
SUS-2178 | remove references from ParserFirstCallInit hook handlers

macbre committed:
SUS-2178 | $parser does not need to be passed by references to a hook

macbre committed:
More works on hooks and passing by reference

macbre committed:
No need to pass parser instance by reference deliberately

WikiaTech commented on extensions/3rdparty/HasCategory/HasCategory.php:
```diff
@@ -94,7 +94,7 @@ function IfhascatPfObj( &$parser, $frame, $args ) {
 	 * If yes return $then (default: 1) else return $else (default: 0).
 	 */
-	function IfhascatPf( &$parser, $page = '', $category = '', $then = '1', $else = '0' ) {
+	function IfhascatPf( $parser, $page = '', $category = '', $then = '1', $else = '0' ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/RSHighscores/RSHighscores.body.php:
```diff
@@ -113,7 +113,7 @@ private static function parseHiscores( $data, $skill, $type ) {
 	 * @return string
 	 */
-	public static function renderHiscores( &$parser, $hs = 'rs3', $player = '', $skill = -1, $type = 1 ) {
+	public static function renderHiscores( $parser, $hs = 'rs3', $player = '', $skill = -1, $type = 1 ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/ArticleMetaDescription/ArticleMetaDescription.php:
```diff
@@ -35,7 +35,7 @@
  * @return bool
  */
-function wfArticleMetaDescription( &$out, &$text ) {
+function wfArticleMetaDescription( OutputPage $out, &$text ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$text". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/Variables/Variables.php:
```diff
@@ -177,7 +177,7 @@ static function pfObj_var( Parser &$parser, $frame, $args) {
 	
-	static function pf_var_final( Parser &$parser, $varName, $defaultVal = '' ) {
+	static function pf_var_final( Parser $parser, $varName, $defaultVal = '' ) {
 		$varStore = self::get( $parser );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$varStore" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on extensions/Variables/Variables.php:
```diff
@@ -146,25 +146,25 @@ public static function getDir() {
 		
-    static function pfObj_var( Parser &$parser, $frame, $args) {
+    static function pfObj_var( Parser $parser, $frame, $args) {
 		$varName = trim( $frame->expand( $args[0] ) ); // first argument expanded already but lets do this anyway
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/3rdparty/DynamicFunctions/DynamicFunctions.php:
```diff
@@ -97,7 +97,7 @@ function rand( &$parser, $a = 0, $b = 1 )
-    function skin( &$parser )
+    function skin( $parser )
     {
         global $wgUser, $wgRequest;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on extensions/3rdparty/DynamicFunctions/DynamicFunctions.php:
```diff
@@ -66,7 +66,7 @@ class ExtDynamicFunctions
-    function arg( &$parser, $name = '', $default = '' )
+    function arg( $parser, $name = '', $default = '' )
     {
         global $wgRequest;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

WikiaTech commented on extensions/DPLforum/DPLforum_body.php:
```diff
@@ -90,7 +90,7 @@ function get( $name, $value = null, $parser = null ) {
 	}
 
-	function link( &$parser, $count, $page = '', $text = '' ) {
+	function link( $parser, $count, $page = '', $text = '' ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/ContentFeeds/ContentFeeds.php:
```diff
@@ -46,7 +46,7 @@ function wfContentFeedsInit() {
 	 */
 	//$wgAjaxExportList[] = '';
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1533,7 +1533,7 @@ public static function fixCategory($cat) {
 
 // reset everything; some categories may have been fixed, however via  fixcategory=
-    public static function endReset( &$parser, $text ) {
+    public static function endReset( $parser, $text ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$text". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1550,7 +1550,7 @@ public static function endReset( &$parser, $text ) {
     }
 
-    public static function endEliminate( &$parser, &$text ) {
+    public static function endEliminate( $parser, &$text ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$text". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1391,7 +1391,7 @@ public static function dplParserFunction(&$parser) {
     }
 
-    public static function dplNumParserFunction(&$parser, $text='') {
+    public static function dplNumParserFunction($parser, $text='') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1409,7 +1409,7 @@ public static function dplNumParserFunction(&$parser, $text='') {
     } 
 
-    public static function dplVarParserFunction(&$parser, $cmd) {
+    public static function dplVarParserFunction($parser, $cmd) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1425,7 +1425,7 @@ private static function isRegexp ($needle) {
     }
 
-    public static function dplReplaceParserFunction(&$parser, $text, $pat, $repl='') {
+    public static function dplReplaceParserFunction($parser, $text, $pat, $repl='') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1436,12 +1436,12 @@ public static function dplReplaceParserFunction(&$parser, $text, $pat, $repl='')
     } 
 
-    public static function dplMatrixParserFunction(&$parser, $name, $yes, $no, $flip, $matrix ) {
+    public static function dplMatrixParserFunction($parser, $name, $yes, $no, $flip, $matrix ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "dplMatrixParserFunction" is 24 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1425,7 +1425,7 @@ private static function isRegexp ($needle) {
 
-    public static function dplReplaceParserFunction(&$parser, $text, $pat, $repl='') {
+    public static function dplReplaceParserFunction($parser, $text, $pat, $repl='') {
 		if ($text=='' || $pat=='') return '';
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1409,7 +1409,7 @@ public static function dplNumParserFunction(&$parser, $text='') {
-    public static function dplVarParserFunction(&$parser, $cmd) {
+    public static function dplVarParserFunction($parser, $cmd) {
 		$args = func_get_args();
         if      ($cmd=='set')     return DPLVariables::setVar($args);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on extensions/DynamicPageList/DPLSetup.php:
```diff
@@ -1409,7 +1409,7 @@ public static function dplNumParserFunction(&$parser, $text='') {
+    public static function dplVarParserFunction($parser, $cmd) {
 		$args = func_get_args();
         if      ($cmd=='set')     return DPLVariables::setVar($args);
         else if ($cmd=='default') return DPLVariables::setVarDefault($args);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -203,7 +203,7 @@ static function renderFormRedLink ( &$parser ) {
-	static function renderQueryFormLink ( &$parser ) {
+	static function renderQueryFormLink ( $parser ) {
 		$params = func_get_args();
 		array_shift( $params ); // We don't need the parser.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -465,7 +465,7 @@ static function renderArrayMapTemplate( &$parser, $frame, $args ) {
 
 
-	static function renderAutoEdit( &$parser ) {
+	static function renderAutoEdit( $parser ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "renderAutoEdit" is 23 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -212,7 +212,7 @@ static function renderQueryFormLink ( &$parser ) {
 	}
 
-	static function renderFormInput( &$parser ) {
+	static function renderFormInput( $parser ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "renderFormInput" has 175 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -194,7 +194,7 @@ static function renderFormLink ( &$parser ) {
-	static function renderFormRedLink ( &$parser ) {
+	static function renderFormRedLink ( $parser ) {
 		$params = func_get_args();
 		array_shift( $params ); // We don't need the parser.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -212,7 +212,7 @@ static function renderQueryFormLink ( &$parser ) {
-	static function renderFormInput( &$parser ) {
+	static function renderFormInput( $parser ) {
 		$params = func_get_args();
 		array_shift( $params ); // don't need the parser
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -601,7 +601,7 @@ static function renderAutoEdit( &$parser ) {
 	}
 
-	static function createFormLink( &$parser, $params, $parserFunctionName ) {
+	static function createFormLink( $parser, $params, $parserFunctionName ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "createFormLink" is 32 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "createFormLink" has 163 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

WikiaTech commented on extensions/SemanticForms/includes/SF_ParserFunctions.php:
```diff
@@ -185,7 +185,7 @@ static function renderDefaultform( &$parser ) {
-	static function renderFormLink ( &$parser ) {
+	static function renderFormLink ( $parser ) {
 		$params = func_get_args();
 		array_shift( $params ); // We don't need the parser.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented on includes/wikia/Wikia.php:
```diff
@@ -2335,7 +2335,7 @@ static function onLocalFilePurgeThumbnailsUrls( LocalFile $file, Array &$urls )
 	 * return false stops permissions processing and we are totally decided (nothing later can override)
 	 */
-	static function canEditInterfaceWhitelist ( &$title, &$wgUser, $action, &$result ) {
+	static function canEditInterfaceWhitelist ( Title $title, User $wgUser, $action, &$result ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/wikia/CategoryGalleries/CategoryGalleriesHelper.class.php:
```diff
@@ -62,7 +62,7 @@ static public function onLanguageGetMagic(&$magicWords, $langCode){
 		 * Hook entry for removing the magic words from displayed text
 		 */
-		static public function onInternalParseBeforeLinks(&$parser, &$text, &$strip_state) {
+		static public function onInternalParseBeforeLinks($parser, &$text, &$strip_state) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$strip_state". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

mszabo-wikia commented:
replaced by https://github.com/Wikia/app/pull/13444

WikiaTech commented on extensions/HeaderTabs/HeaderTabs_body.yui.php:
```diff
@@ -27,7 +27,7 @@ public static function tag( $input, $args, $parser ) {
 	}
 
-	public static function replaceFirstLevelHeaders( &$parser, &$text ) {
+	public static function replaceFirstLevelHeaders( $parser, &$text ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/HeaderTabs/HeaderTabs_body.jq.php:
```diff
@@ -17,7 +17,7 @@ public static function tag( $input, $args, $parser ) {
 	}
 
-	public static function replaceFirstLevelHeaders( &$parser, &$text ) {
+	public static function replaceFirstLevelHeaders( $parser, &$text ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "replaceFirstLevelHeaders" is 38 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "replaceFirstLevelHeaders" has 243 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

WikiaTech commented on extensions/NumberFormat/NumberFormat.php:
```diff
@@ -27,7 +27,7 @@ function number_format_Setup( Parser $parser )	 {
-	public static function number_format_Render( &$parser ) {
+	public static function number_format_Render( $parser ) {
 		wfProfileIn(__METHOD__);
 		// {{#number_format:number|decimals|dec_point|thousands_sep}}
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech commented on extensions/NumberFormat/NumberFormat.php:
```diff
@@ -27,7 +27,7 @@ function number_format_Setup( Parser $parser )	 {
 class NumberFormart {
 
-	public static function number_format_Render( &$parser ) {
+	public static function number_format_Render( $parser ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "number_format_Render" is 21 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -287,7 +287,7 @@ function arraysize( &$parser, $arrayid) {
 *   note it is extended to support regular expression match and index
 */
-    function arraysearch( &$parser, $arrayid, $needle, $index=0, $yes=null, $no=null) {
+    function arraysearch( $parser, $arrayid, $needle, $index=0, $yes=null, $no=null) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -575,7 +575,7 @@ function arrayunion( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
 */
 
-    function arraydiff( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arraydiff( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -575,7 +575,7 @@ function arrayunion( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
 
-    function arraydiff( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arraydiff( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
         if (!isset($arrayid_new) )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -481,7 +481,7 @@ function is_non_empty($var){
 *    see: http://www.php.net/manual/en/function.array-slice.php
 */
-    function arrayslice( &$parser, $arrayid_new , $arrayid , $offset, $length='') {
+    function arrayslice( $parser, $arrayid_new , $arrayid , $offset, $length='') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -481,7 +481,7 @@ function is_non_empty($var){
 */
-    function arrayslice( &$parser, $arrayid_new , $arrayid , $offset, $length='') {
+    function arrayslice( $parser, $arrayid_new , $arrayid , $offset, $length='') {
         if (!isset($arrayid_new) )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -355,7 +355,7 @@ function arraysearch( &$parser, $arrayid, $needle, $index=0, $yes=null, $no=null
 *    {{#arrayreset:arrayid1,arrayid2,...arrayidn}}
 */
-   function arrayreset( &$parser, $arrayids) {
+   function arrayreset( $parser, $arrayids) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -547,7 +547,7 @@ function arrayintersect( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
 */
 
-    function arrayunion( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arrayunion( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -547,7 +547,7 @@ function arrayintersect( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
 
-    function arrayunion( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arrayunion( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
         if (!isset($arrayid_new) )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -519,7 +519,7 @@ function arrayslice( &$parser, $arrayid_new , $arrayid , $offset, $length='') {
 *   See: http://www.php.net/manual/en/function.array-intersect.php
 */
-    function arrayintersect( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arrayintersect( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -519,7 +519,7 @@ function arrayslice( &$parser, $arrayid_new , $arrayid , $offset, $length='') {
 */
-    function arrayintersect( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arrayintersect( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
         if (!isset($arrayid_new) )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -265,7 +265,7 @@ function arrayindex( &$parser, $arrayid , $index , $options='') {
 *  See: http://www.php.net/manual/en/function.count.php
 */
-    function arraysize( &$parser, $arrayid) {
+    function arraysize( $parser, $arrayid) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -142,7 +142,7 @@ class ArrayExtension {
 * see also: http://us2.php.net/manual/en/function.preg-split.php
 */
-    function arraydefine( &$parser, $arrayid, $value='', $delimiter = '/\s*,\s*/', $options = '') {
+    function arraydefine( $parser, $arrayid, $value='', $delimiter = '/\s*,\s*/', $options = '') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -142,7 +142,7 @@ class ArrayExtension {
 */
-    function arraydefine( &$parser, $arrayid, $value='', $delimiter = '/\s*,\s*/', $options = '') {
+    function arraydefine( $parser, $arrayid, $value='', $delimiter = '/\s*,\s*/', $options = '') {
         if (!isset($arrayid))
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -240,7 +240,7 @@ function arrayprintObj(  &$parser, $frame, $args ) {
 *   {{#arrayindex:arrayid|index}}
 */
-    function arrayindex( &$parser, $arrayid , $index , $options='') {
+    function arrayindex( $parser, $arrayid , $index , $options='') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -442,7 +442,7 @@ function arraysort( &$parser, $arrayid , $sort = 'none') {
 *  this merge differs from array_merge of php because it merges values.
 */
-    function arraymerge( &$parser, $arrayid_new, $arrayid1, $arrayid2='' ) {
+    function arraymerge( $parser, $arrayid_new, $arrayid1, $arrayid2='' ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -411,7 +411,7 @@ function arrayunique( &$parser, $arrayid ) {
 *          http://us3.php.net/manual/en/function.array-reverse.php
 */
-    function arraysort( &$parser, $arrayid , $sort = 'none') {
+    function arraysort( $parser, $arrayid , $sort = 'none') {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -442,7 +442,7 @@ function arraysort( &$parser, $arrayid , $sort = 'none') {
 */
-    function arraymerge( &$parser, $arrayid_new, $arrayid1, $arrayid2='' ) {
+    function arraymerge( $parser, $arrayid_new, $arrayid1, $arrayid2='' ) {
         if (!isset($arrayid_new) )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

WikiaTech commented on extensions/3rdparty/ArrayExtension/ArrayExtension.php:
```diff
@@ -380,7 +380,7 @@ function arrayreset( &$parser, $arrayids) {
 *   see: http://www.php.net/manual/en/function.array-unique.php
 */
-    function arrayunique( &$parser, $arrayid ) {
+    function arrayunique( $parser, $arrayid ) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

WikiaTech commented:
SonarQube analysis reported 290 issues
* ![MAJOR][MAJOR] 222 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 55 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ArrayExtension.php#L32](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L32): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L294](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L294): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L302](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L302): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L310](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L310): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L335](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L335): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L341](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L341): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L420](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L420): Add a "case default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L493](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L493): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L506](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L506): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [ArrayExtension.php#L602](https://github.com/Wikia/app/blob/59f84d09c0daa2eb56cca6e968f2221740bc5f45/extensions/3rdparty/ArrayExtension/ArrayExtension.php#L602): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
